deltachat-ios/Helper/ImageFormat.swift

```diff
@@ -67,4 +68,35 @@ extension ImageFormat {
         return loadImageFrom(data: imageData)
     }
 
+    public static func saveImage(image: UIImage) -> String? {
+        if image.sd_isAnimated,
+           let data = image.sd_imageData() {
+            let format = ImageFormat.get(from: data)
+            if format != .unknown {
+                return ImageFormat.saveImage(data: data, suffix: format.rawValue)
+            }
+        }
+        let suffix = image.isTransparent() ? "png" : "jpg"
+        guard let data = image.isTransparent() ? image.pngData() : image.jpegData(compressionQuality: 1.0) else {
+            return nil
+        }
+
+        return saveImage(data: data, suffix: suffix)
+    }
+
+    public static func saveImage(data: Data, suffix: String) -> String? {
```

Comment:
so, data contains the already encoded image data, not an rgb-bitarray? and image-(re)encoding does not take place in this function? maybe add a comment on that and/or name data to encoded_data.